jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -185,47 +198,55 @@ ecma_builtin_global_object_parse_int (ecma_value_t this_arg __attr_unused___, /*
       /* 10. */
       if (strip_prefix)
       {
-        if (end - start >= 2
-            && utf8_string_buff[start] == LIT_CHAR_0
-            && (utf8_string_buff[start + 1] == LIT_CHAR_LOWERCASE_X
-                || utf8_string_buff[start + 1] == LIT_CHAR_UPPERCASE_X))
+        if (end.offset - start.offset >= 2 && current == LIT_CHAR_0)
         {
-          start += 2;
+          ecma_char_t next = lit_utf8_iterator_peek_next (&iter);
```

Comment:
In this case it is not needed, as the offset check in the `if` statement above already checked that there are at least two characters, so we are not at the end of the string yet.